### PR TITLE
[OSDEV-1045] Change color for claimed facility in merge screen

### PR DESCRIPTION
**Pull request by Innavin369**
[OSDEV-1045](https://opensupplyhub.atlassian.net/browse/OSDEV-1045) Data Moderation. Change the color of the facility panel for claimed facilities in the Merge moderation screens.

* moved colors to constants
* moved titles for Dashboard Facility Card to constants
* added `highlightBackground` flag to highlight panels for claimed facilities only on the Dashboard Facility Merge screen
* added unit and component tests

[OSDEV-1045]: https://opensupplyhub.atlassian.net/browse/OSDEV-1045?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by VadimKovalenkoSNF — CHANGES_REQUESTED** [review]
Can you add frontend unit tests, please?

**Review by mazursasha1990 — CHANGES_REQUESTED** [review]

**Review by mazursasha1990 — CHANGES_REQUESTED** [review]
After reviewing your PR, I noticed that the provided solution does not meet the acceptance criteria. Specifically, it incorrectly assigns a green background to facilities that have a claim with 'PENDING' or 'DENIED' status. 
Additionally, I believe there could be a different approach to this implementation. Please contact me directly for a more detailed explanation.

File: doc/release/RELEASE-NOTES.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
 *
 
 ### What's new
+* [OSDEV-1045](https://opensupplyhub.atlassian.net/browse/OSDEV-1045) - Data Moderation. Change the color of the facility panel for claimed facilities in the Merge moderation screens.
```
[the remark this answers is not in the document]
Comment (Innavin369) [reply doc/release/RELEASE-NOTES.md:34]: done

File: src/react/src/components/DashboardUpdateFacilityLocation.jsx [hunk 2]
```diff
@@ -86,7 +91,10 @@ function DashboardUpdateFacilityLocation({
                     fetching={targetFetching}
                     error={targetError}
                     handleEnterKeyPress={fetchTargetFacilityOnEnterKeyPress}
-                    title="Facility"
+                    title={dashboardFacilityCardTitles.facility}
+                    highlightBackground={dashboardFacilityMergeCards.includes(
```
Comment (mazursasha1990) [inline src/react/src/components/DashboardUpdateFacilityLocation.jsx:95]: Don't pass highlightBackground if you don't need to change the background for the card in the DashboardDeleteFacility component.

File: src/react/src/util/constants.jsx [hunk 3]
```diff
@@ -1289,3 +1289,16 @@ export const componentsWithErrorMessage = Object.freeze({
...
+    facilityToDelete: 'Facility to delete',
+    targetFacilityToMerge: 'Target facility for merge',
+    facilityToMergeIntoTarget: 'Facility to merge into target',
+    facility: 'facility',
+});
+
+export const dashboardFacilityMergeCards = Object.freeze([
+    'Target facility for merge',
+    'Facility to merge into target',
+]);
```
Comment (mazursasha1990) [inline src/react/src/util/constants.jsx:1304]: I believe these changes aren't necessary and you don't need to add them.

File: doc/release/RELEASE-NOTES.md [hunk 4]
```diff
@@ -31,6 +31,7 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
 *
 
 ### What's new
+* [OSDEV-1045](https://opensupplyhub.atlassian.net/browse/OSDEV-1045) - The color of the facility panel for claimed facilities in the Merge moderation screens has been changed to green. Added flag `highlightBackground` for different screens to highlight panel only on the Dashboard Merge Facilities screen.
```
Comment (mazursasha1990) [inline doc/release/RELEASE-NOTES.md:34]: Please update the description after implementing the changes according to the comments I provided below.

**Review by mazursasha1990 — CHANGES_REQUESTED** [review]

File: src/react/src/__tests__/components/DashboardFacilityCard.test.js [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+
+    it('renders the claimed facility card with green background', () => {
+        const { container } = renderComponent({highlightBackground: true,
+            data: {
+                ...defaultProps.data,
+                properties: {
+                    ...defaultProps.data.properties,
+                    is_claimed: true,
+                },
+            },});
```
Comment (mazursasha1990) [inline src/react/src/__tests__/components/DashboardFacilityCard.test.js:252]: I believe that moving the complex prop object into a variable can improve the readability and maintainability of the test.

    const claimedFacilityProps = {
        highlightBackground: true,
        data: {
            ...defaultProps.data,
            properties: {
                ...defaultProps.data.properties,
                is_claimed: true,
            },
        },
    };

    const { container } = renderComponent(claimedFacilityProps);

File: doc/release/RELEASE-NOTES.md [hunk 6]
```diff
@@ -32,6 +32,7 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
 *
 
 ### What's new
+* [OSDEV-1045](https://opensupplyhub.atlassian.net/browse/OSDEV-1045) - The color of the facility panel for claimed facilities in the Merge moderation screen has been changed to green. Added flag `highlightBackground` to highlight panel only on the Dashboard Merge Facilities screen.
```
Comment (mazursasha1990) [inline doc/release/RELEASE-NOTES.md:35]: > Added flag `highlightBackground` to highlight panel only on the Dashboard Merge Facilities screen.

I think it's not relevant to the "What's new" section.

Also, please describe the changes in the FacilityIndexDetailsSerializer.

**Review by mazursasha1990 — CHANGES_REQUESTED** [review]

**Review by VadimKovalenkoSNF — APPROVED** [review]
Approved from my side.

**Review by roman-stolar — DISMISSED** [review]
In general LGTM, please take a look to @mazursasha1990 comment.

**Review by mazursasha1990 — APPROVED** [review]
LGTM

**Comment by roman-stolar** [comment]
@Innavin369 Please check django tests, I will approve when it passed on CI.

**Review by VadimKovalenkoSNF — APPROVED** [review]
Approved

**Review by mazursasha1990 — APPROVED** [review]
LGTM

**Pull request merged**